android-net/src/main/java/li/vin/net/Device.java

```diff
@@ -262,7 +266,16 @@ public void onMessage(BufferedSource payload, WebSocket.PayloadType type)
                     try {
                       String payloadStr = payload.readString(UTF8);
                       if (payloadStr != null && !payloadStr.isEmpty()) {
-                        subscriber.onNext(gson.fromJson(payloadStr, StreamMessage.class));
+                        StreamMessage streamMessage = gson.fromJson(payloadStr, StreamMessage.class);
+
+                        synchronized (bearingCalculator){
+                          if(streamMessage.coord() != null){
```

Comment:
Only synchronize around as much code as you absolutely have to - put synchronized block inside of if null check block instead of other way around.